modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/gateway/CommercePaymentGatewayImpl.java

```diff
@@ -94,6 +111,20 @@ public CommercePaymentEntry authorize(
 				_createServiceContext(currentUser));
 		}
 
+		if (StringUtil.equals(
+				commercePaymentEntry.getClassName(),
+				CommerceOrder.class.getName())) {
+
+			PermissionThreadLocal.setPermissionChecker(
+				PermissionCheckerFactoryUtil.create(currentUser));
+
+			updateOrderPaymentStatus(
```

Comment:
I would decouple this from the PaymentGateway. I would create a messageListener on the CommercePaymentEntry status change, that will change the order paymentStatus if we are handling a CommerceOrder.